From a9f378b283a3a60570f8c8242942c8ef23295ce7 Mon Sep 17 00:00:00 2001
From: Takuto Ikuta <tikuta@chromium.org>
Date: Thu, 7 Jul 2022 03:16:54 +0000
Subject: [PATCH 066/110] testing: support python3 in
 headless_python_unittests.py for M102
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

(cherry picked from commit d0a8c00a16849a549b80ec49146f5c7db42586df)

No-Presubmit: true
Bug: 1340449
Change-Id: I364fce6f0b86f0129e8f9e94de744c59dea115ac
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3734707
Reviewed-by: Sami Kyöstilä <skyostil@chromium.org>
Reviewed-by: Brian Sheedy <bsheedy@chromium.org>
Commit-Queue: Sami Kyöstilä <skyostil@chromium.org>
Auto-Submit: Takuto Ikuta <tikuta@chromium.org>
Cr-Original-Commit-Position: refs/heads/main@{#1019917}
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3739880
Commit-Queue: Brian Sheedy <bsheedy@chromium.org>
Cr-Commit-Position: refs/branch-heads/5005@{#1225}
Cr-Branched-From: 5b4d9450fee01f821b6400e947b3839727643a71-refs/heads/main@{#992738}
---
 .../lib/browser/devtools_api/client_api_generator_unittest.py | 4 ++--
 testing/scripts/headless_python_unittests.py                  | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/headless/lib/browser/devtools_api/client_api_generator_unittest.py b/headless/lib/browser/devtools_api/client_api_generator_unittest.py
index 9e1ca6568402..cd49df0209c5 100755
--- a/headless/lib/browser/devtools_api/client_api_generator_unittest.py
+++ b/headless/lib/browser/devtools_api/client_api_generator_unittest.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # Copyright 2016 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
@@ -15,7 +15,7 @@ class ClientApiGeneratorTest(unittest.TestCase):
 
   def test_ArgumentParsing(self):
     with tempfile.NamedTemporaryFile() as f:
-      f.write('{"foo": true}')
+      f.write(b'{"foo": true}')
       f.flush()
       json_api, output_dir = client_api_generator.ParseArguments([
           '--protocol', f.name, '--output_dir', 'out'])
diff --git a/testing/scripts/headless_python_unittests.py b/testing/scripts/headless_python_unittests.py
index 881cff4a6ebb..14a7a427c3f4 100755
--- a/testing/scripts/headless_python_unittests.py
+++ b/testing/scripts/headless_python_unittests.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env vpython3
 # Copyright 2016 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
@@ -36,7 +36,7 @@ def main_run(args):
   valid = bool(rc <= common.MAX_FAILURES_EXIT_STATUS and
                ((rc == 0) or failures))
   common.record_local_script_results(
-      'headless_python_unittests', args.output, failures.keys(), valid)
+      'headless_python_unittests', args.output, list(failures.keys()), valid)
 
   return rc
 
-- 
2.38.1

